odk/src/main/java/org/motechproject/odk/domain/builder/FormInstanceBuilder.java

```diff
@@ -0,0 +1,160 @@
+package org.motechproject.odk.domain.builder;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.joda.time.DateTime;
+import org.motechproject.odk.constant.FieldTypeConstants;
+import org.motechproject.odk.domain.FormDefinition;
+import org.motechproject.odk.domain.FormElement;
+import org.motechproject.odk.domain.FormInstance;
+import org.motechproject.odk.domain.FormValue;
+import org.motechproject.odk.domain.FormValueDateTime;
+import org.motechproject.odk.domain.FormValueDouble;
+import org.motechproject.odk.domain.FormValueGroup;
+import org.motechproject.odk.domain.FormValueInteger;
+import org.motechproject.odk.domain.FormValueString;
+import org.motechproject.odk.domain.FormValueStringList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Builder class for {@link org.motechproject.odk.domain.FormInstance}
+ */
+public class FormInstanceBuilder {
+
+    private FormDefinition formDefinition;
+    private Map<String, Object> params;
+    private String instanceId;
+
+
+    public FormInstanceBuilder(FormDefinition formDefinition, Map<String, Object> params, String instanceId) {
+        this.formDefinition = formDefinition;
+        this.params = params;
+        this.instanceId = instanceId;
+    }
+
+    /**
+     * Builds a {@link FormInstance}
+     *
+     * @return {@link FormInstance}
+     */
+    public FormInstance build() {
+        List<FormValue> formValues = new ArrayList<>();
+        List<FormElement> formElements = formDefinition.getFormElements();
+
+        for (FormElement formElement : formElements) {
+            Object value = params.get(formElement.getName());
+
+            if (value != null && !(formElement.getType().equals(FieldTypeConstants.SELECT)
+                    && ((List<String>) value).size() == 0)) {
```

Comment:
perhaps it would help to check if the value is a list using instanceof, then throw an exception which will explain the error more. Currently this will be a class cast